dkc/core/admin/folder.py

```diff
@@ -9,3 +11,15 @@ class FolderAdmin(admin.ModelAdmin):
     list_display_links = ['id', 'name']
 
     search_fields = ['name', 'id']
+
+    @transaction.atomic
+    def save_model(self, request, obj: Folder, form, change: bool):
+        if not change:
+            if obj.parent:
+                obj.tree = obj.parent.tree
+            else:
+                obj.tree = Tree.objects.create(quota=request.user.quota)
+                obj.tree.grant_permission(
+                    PermissionGrant(user_or_group=request.user, permission=Permission.admin)
```

Comment:
What if we want to grant this to another user or not grant it at all?

Reply:
I'm fine to omit this, this is just an attempt to not have a 500 when creating a folder through the admin page.

Reply:
What will 500? I would expect a 403 somewhere at worst.

Though can't an admin already do anything, even without explicit permission?

Reply:
On current master, trying to create a new folder via the admin page gives a 500 because it attempts to save a folder with no tree attached. @danlamanna ran into this bug earlier today. We could maybe try to preserve purity by making people manually create a Tree object and then attach a folder to it, but personally I'd rather they always get created together atomically.

Reply:
Sorry, I just mean is it valid to omit `obj.tree.grant_permission`?

Reply:
Yes, I believe that's valid.